ssl/record/ssl3_record_tls13.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "../ssl_locl.h"
+#include "record_locl.h"
+
+/*-
+ * tls13_enc encrypts/decrypts |n_recs| in |recs|.
+ *
+ * Returns:
+ *    0: (in non-constant time) if the record is publically invalid (i.e. too
```

Comment:
Since none of these will ever set `read_hash` and thus never hit the MAC code and also will never hit the constant-time CBC business, I believe the 0 / -1 distinction and constant-time worries aren't meaningful. (Hopefully once the refactoring in #1438 happens, it'll be less confusing.)